app/services/cache_service.rb

```diff
@@ -6,12 +6,12 @@ def prefetch_listings(opts = {})
     Rails.logger.info('CacheService Started')
     # Marshal.load(Marshal.dump(cached_listings)) is a hack to make a deep clone of cached_listings
     # so that prev_cached_listings isn't pointing to the same object as fresh_listings
-    cached_listings = Force::ListingService.listings(subset: 'browse')
+    cached_listings = Force::ListingService.listings(subset: 'browse', cache_only: true)
     @prev_cached_listings = Marshal.load(Marshal.dump(cached_listings))
 
     @fresh_listings = Force::ListingService.listings(subset: 'browse', force: true)
 
-    if opts[:refresh_all]
+    if opts[:refresh_all] || @prev_cached_listings.nil?
```

Comment:
Question: I missed this line when I reviewed the first time. This is the code that will addresses the main issue right? I'm still not clear on what the new `cache_only` argument does though.

Reply:
@prev_cached_listings won't return as nil without the cache_only functionality